src/crud.py

```
@@ -672,44 +863,74 @@ async def promote_prompt_version(db: AsyncSession, version_id: int) -> bool:
+        logging.info(f"[CRUD] Created EvaluationRun object with id: {db_run.id}")
         
-        if version:
-            run_prompt = EvaluationRunPrompt(
-                evaluation_run_id=db_run.id,
-                prompt_version_id=version.id,
-                label=config.label
+        # Append datasets BEFORE flush/refresh
+        for dataset_id in run.dataset_ids:
+            logging.info(f"[CRUD] Adding dataset_id: {dataset_id} to run {db_run.id}")
+            dataset = await get_dataset(db, dataset_id)
+            logging.info(f"[Crud]!!Dataset Found")
```

Ruff (0.11.9)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)